dev/reference/alert-rules.md

```diff
@@ -0,0 +1,1276 @@
+
+#### `TiDB_server_event_error`
+
+* Alert rule:
+
+    `increase(tidb_server_server_event{type=~"server_start|server_hang"}[15m]) > 0`
+
+* Description:
+
+    The number of events that happen in the TiDB service. An alert is triggered when the following events happen:
+
+    1. start: The TiDB service starts.
+    2. hang: When a critical-level event (currently there’s only one scenario: TiDB cannot write binlog) happens, TiDB enters the `hang` mode and waits to be killed manually.
+
+* Solution:
+
+    * Restart TiDB to recover the service.
+    * Check whether the TiDB Binlog service is normal.
+
+#### `TiDB_tikvclient_backoff_count`
+
+* Alert rule:
+
+    `increase(tidb_tikvclient_backoff_count[10m]) > 10`
+
+* Description:
+
+    The number of retries when TiDB fails to access TiKV. When the retry times is over 10 in 10 minutes, an alert is triggered.
+
+* Solution:
+
+    View the monitoring status of TiKV.
+
+#### `TiDB_monitor_time_jump_back_error`
+
+* Alert rule:
+
+    `increase(tidb_monitor_time_jump_back_total[10m]) > 0`
+
+* Description:
+
+    When the time of the machine that holds TiDB rewinds, an alert is triggered.
+
+* Solution:
+
+    Troubleshoot the NTP configurations.
+
+#### `TiDB_ddl_waiting_jobs`
+
+* Alert rule:
+
+    `sum(tidb_ddl_waiting_jobs) > 5`
+
+* Description:
+
+    When the number of DDL tasks pending for execution in TiDB exceeds 5, an alert is triggered.
+
+* Solution:
+
+    Check whether there is any time-consuming `add index` operation that is being executed by running `admin show ddl`.
+
+## PD alert rules
+
+This section gives the alert rules for the PD component.
+
+### Emergency-level alerts
+
+Emergency-level alerts are often caused by a service or node failure. Manual intervention is required immediately.
+
+#### `PD_cluster_offline_tikv_nums`
+
+* Alert rule:
+
+    `sum(pd_cluster_status{type="store_down_count"}) > 0`
+
+* Description:
+
+    PD has not received a TiKV heartbeat for a long time (the default configuration is 30 minutes).
+
+* Solution:
+
+    * Check whether the TiKV process is normal, the network is isolated or the load is too high, and recover the service as much as possible.
+    * If the TiKV cannot be recovered, you can make it offline.
+
+### Critical-level alerts
+
+For the critical-level alerts, a close watch on the abnormal metrics is required.
+
+#### `PD_etcd_write_disk_latency`
+
+* Alert rule:
+
+    `histogram_quantile(0.99, sum(rate(etcd_disk_wal_fsync_duration_seconds_bucket[1m])) by (instance,job,le)) > 1`
+
+* Description:
+
+    etcd writes data to disk with a slower speed than normal. It might lead to PD leader timeout or failure to store TSO on disk in time, which will shut down the service of the entire cluster.
+
+* Solution:
+
+    * Find the cause of slow writes. It might be other services that overload the system. You can check whether PD itself occupies a large amount of CPU or IO resources.
```

Comment:
```suggestion
    * Find the cause of slow writes. It might be other services that overload the system. You can check whether PD itself occupies a large amount of CPU or I/O resources.
```